gravitee-apim-console-webui/src/management/api/portal/user-group-access/members/api-portal-members.component.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { fakeApi } from '../../../../../entities/api/Api.fixture';
+import { Api, Member } from '../../../../../entities/api';
+import { UsersService } from '../../../../../services-ngx/users.service';
+import { UIRouterStateParams } from '../../../../../ajs-upgraded-providers';
+import { RoleService } from '../../../../../services-ngx/role.service';
+import { Role } from '../../../../../entities/role/role';
+import { fakeRole } from '../../../../../entities/role/role.fixture';
+
+describe('ApiPortalMembersComponent', () => {
+  let fixture: ComponentFixture<ApiPortalMembersComponent>;
+  let httpTestingController: HttpTestingController;
+  let harness: ApiPortalMembersHarness;
+  const apiId = 'apiId';
+  const roles: Role[] = [fakeRole({ name: 'PRIMARY_OWNER' }), fakeRole({ name: 'OWNER' }), fakeRole({ name: 'USER' })];
+
+  beforeEach(async () => {
+    TestBed.configureTestingModule({
+      imports: [NoopAnimationsModule, GioHttpTestingModule, ApiPortalUserGroupModule],
+      providers: [
+        { provide: UIRouterStateParams, useValue: { apiId } },
+        { provide: UsersService, useValue: { getUserAvatar: () => 'avatar' } },
+        { provide: RoleService, useValue: { list: () => of(roles) } },
+      ],
+    });
+
+    fixture = TestBed.createComponent(ApiPortalMembersComponent);
+    httpTestingController = TestBed.inject(HttpTestingController);
+
+    harness = await TestbedHarnessEnvironment.harnessForFixture(fixture, ApiPortalMembersHarness);
+
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    httpTestingController.verify();
+  });
+
+  describe('API member change notification', () => {
+    it('should uncheck box when notifications are off', async () => {
+      const api = fakeApi({ id: apiId, disable_membership_notifications: true });
+      expectApiGetRequest(api);
+      expectApiMembersGetRequest();
+
+      expect(await harness.isNotificationsCheckboxChecked()).toEqual(false);
+    });
+
+    it('should show save bar when checkbox is toggles', async () => {
+      const api = fakeApi({ id: apiId, disable_membership_notifications: false });
+      expectApiGetRequest(api);
+      expectApiMembersGetRequest();
+
+      expect(await harness.isNotificationsCheckboxChecked()).toEqual(true);
+      await harness.toggleNotificationCheckbox();
+
+      expect(await harness.isSaveBarVisible()).toEqual(true);
+    });
+
+    it('should save notifications modifications when clicking on save bar', async () => {
+      const api = fakeApi({ id: apiId, disable_membership_notifications: false });
+      expectApiGetRequest(api);
+      expectApiMembersGetRequest();
+
+      await harness.toggleNotificationCheckbox();
+
+      await harness.clickOnSave();
+
+      // Setup all http calls for save + new OnInit execution
+      // save
+      expectApiGetRequest(api);
+      const req = httpTestingController.expectOne({ url: `${CONSTANTS_TESTING.env.baseURL}/apis/${apiId}`, method: 'PUT' });
```

Comment:
You can add a check that the new info is sent in the PUT body with `req.request.body`

Reply:
it is done at the end of the test. But I will lift this up for more clarity